libs/blocks/global-navigation/global-navigation.js

```diff
@@ -35,6 +35,79 @@ const loadStyles = (path) => {
 const loadBlock = (path) => import(path)
   .then((module) => module.default);
 
+// signIn, decorateSignIn and decorateButton can be removed if IMS takes over the profile
+const signIn = () => {
+  if (typeof window.adobeIMS?.signIn !== 'function') return;
+
+  window.adobeIMS.signIn();
+};
+
+const decorateSignIn = async ({ rawElem, decoratedElem }) => {
+  const dropdownElem = rawElem.querySelector(':scope > div:nth-child(2)');
+  const signInLabel = await replaceKey('sign-in', getFedsPlaceholderConfig(), 'feds');
+  let signInElem;
+
+  if (!dropdownElem) {
+    signInElem = toFragment`<a href="#" daa-ll="${signInLabel}" class="feds-signIn">${signInLabel}</a>`;
+
+    signInElem.addEventListener('click', (e) => {
+      e.preventDefault();
+      signIn();
+    });
+  } else {
+    signInElem = toFragment`<a href="#" daa-ll="${signInLabel}" class="feds-signIn" role="button" aria-expanded="false" aria-haspopup="true">${signInLabel}</a>`;
+
+    signInElem.addEventListener('click', () => {
+      const isOpen = signInElem.getAttribute('aria-expanded') === 'true';
+
+      if (isOpen) {
+        signInElem.setAttribute('aria-expanded', 'false');
+      } else {
+        signInElem.setAttribute('aria-expanded', 'true');
+      }
+    });
+
+    dropdownElem.classList.add('feds-signIn-dropdown');
+
+    // TODO we don't have a good way of adding config properties to links
+    const dropdownSignIn = dropdownElem.querySelector('[href="https://adobe.com?sign-in=true"]');
+
+    if (dropdownSignIn) {
+      dropdownSignIn.addEventListener('click', (e) => {
+        e.preventDefault();
+        signIn();
+      });
+    }
+
+    decoratedElem.append(dropdownElem);
+  }
+
+  decoratedElem.prepend(signInElem);
+};
+
+const decorateButton = async ({ avatar }) => {
```

Comment:
Outside of the Profile folder context, this is misleading, I'd rename it to `decorateProfileButton` or `decorateProfileTrigger` / `decorateProfileAvatar`, whatever you see fit.